docs/001_develop/03_client-capabilities/023_custom-components/05_directives

```diff
@@ -0,0 +1,291 @@
+        this.data = { message: "Hello, Genesis!" }; # Hardcoded data
+        this.ready = true; # Indicate loading is complete
+    }
+}
+```
+
+Initial State:
+
+- When `<my-button>` is first added to the DOM, it displays "Loading..." for 2 seconds (simulated delay).
+
+After 2 Seconds:
+
+- Once the delay completes, ready is set to true, and the button appears with the text "Data Loaded: Hello, Genesis!".
+
+### Syntax breakdown
+
+```shell
+when(condition, template);
+```
+
+- Condition: A boolean expression that determines if the template should be rendered. If true, the template is added to the DOM; if false, it’s removed.
+- Template: Defines the HTML structure to render when the condition is true.
+
+## whenElse
+
+In the previous example with `when`, we demonstrated how to handle conditional rendering for 'if-else' scenarios by using `when` twice. However, there's a more efficient approach: the `whenElse` directive. With `whenElse`, you can manage if-else conditions in a single directive, avoiding the need to use `when` multiple times.
+
+ <!-- Look into how to implement this -->
+
+### Usage in HTML
+
+## Repeat
+
+The `repeat` directive is used to dynamically render a list of items. This directive is highly useful for rendering lists, especially when you need to manage and display data dynamically.
+
+`<my-button-list>` will display a static list of button labels, each rendered as a button. This example shows how to use repeat to render each item in the list.
+
+```shell
+import { GenesisElement, customElement, observable, html, repeat } from "@genesislcap/web-core";
+
+@customElement({
+    name: "my-button-list",
+    template: html<MyButtonList>`
+        <h2>Button List</h2>
+        <ul>
+            ${repeat(x => x.buttons, html<string>`
+                <li><button>${x => x}</button></li>
+            `)}
+        </ul>
+    `,
+})
+export class MyButtonList extends GenesisElement {
+    @observable buttons: string[] = ["Click Me", "Submit", "Reset"]; // Static list of button labels
+}
+```
+
+- The repeat directive iterates over the buttons array and renders each item as a `<button>` inside an `<li>`.
+- Each button displays the label from the buttons array ("Click Me", "Submit", "Reset").
+  Static List of Labels:
+- The buttons array is initialized with static values for simplicity. Each item in the array is displayed as a button in the rendered HTML.
+
+### Usage in HTML
+
+To test <my-button-list>, add it to your HTML page:
+
+```shell
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <script type="module" src="my-button-list.js"></script>
+</head>
+<body>
+    <my-button-list></my-button-list>
+</body>
+</html>
+```
+
+### Syntax breakdown
+
+```shell
+repeat(expression, itemTemplate, options);
+```
+
+- Expression: Specifies the data source for repeat.
+- Item Template: Defines the HTML structure for each item.
+- Options: Adds control over item positioning and view recycling (optional).
+
```

Comment:
I think we might want to elaborate on repeat a bit to explain some of the more advanced bits it can do. See [here](https://fast.design/docs/1.x/fast-element/using-directives#the-repeat-directive) where it talks about opting into positioning. You probably don't need to go as in depth but we should mention it. Also the view recycling section